app/views/wpcc/your_contributions/new.html.erb

```diff
@@ -35,7 +35,7 @@
       'data-dough-validation-config': '{"showValidationSummary": false}',
       'novalidate': true
     }) do |f| %>
-    <div data-dough-component="SalaryConditions ContributionConditions">
+    <div data-dough-component="SalaryConditions ContributionConditions" data-dough-salary-conditions-config="<%= ::Wpcc::MessagePresenter.opt_in_thresholds.to_json %>">
```

Comment:
I think the data should converted to json outside the view, wdyt?

Reply:
Humm, I am unsure. I did what you requested before ... but I returned to the view because it sounds wrong on that time to me. I thought that the message presenter responsibility is to have a ruby object with the salary frequencies plus the thresholds ordered properly and the view is to convert to format which expected. Wdyt of my views?

Reply:
I thought the message presenter's responsibility included converting the data to a format that the view can use. My understanding is that the view should not perform any logic if possible and since the presenter is already getting and formatting the data, it makes sense that the presenter should also format it for the view.

I dont feel strongly about this, but when I see `.to_json` in a view, it is a code smell to me.

Reply:
Ok, I see. I moved the .to_json to the presenter method.